### PR TITLE
Fix TransactionInfoAPI type

### LINKED ISSUE #6501: gasPrice doesn't exist in web3.eth.getTransaction() return type
`gasPrice` doesn't exist in `web3.eth.getTransaction()` return type

[original link ](https://github.com/web3/web3.js/pull/6492)
#### Expected behavior

`gasPrice` exists in the returning type of `web3.eth.getTransaction()`

#### Actual behavior

The error:
```
 Property 'gasPrice' does not exist on type '{ readonly yParity: string; readonly r: string; readonly s: string; readonly maxFeePerGas: string; readonly maxPriorityFeePerGas: string; readonly accessList: { readonly address?: string | undefined; readonly storageKeys?: string[] | undefined; }[]; ... 12 more ...; readonly transactionIndex?: string | undefined; } ...'.
      Property 'gasPrice' does not exist on type '{ readonly yParity: string; readonly r: string; readonly s: string; readonly maxFeePerGas: string; readonly maxPriorityFeePerGas: string; readonly accessList: { readonly address?: string | undefined; readonly storageKeys?: string[] | undefined; }[]; ... 12 more ...; readonly transactionIndex?: string | undefined; }'.

    const gasPrice = BigInt(transactionDetails.gasPrice);
```

#### Steps to reproduce the behavior

Try to build code:
```ts
const test = async () => {
	const web3 = new Web3();
	const transactionDetails = await web3.eth.getTransaction('0x');
	// here it will throw error that gasPrice is not a property of transaction
	const gasPrice = BigInt(transactionDetails.gasPrice);

	console.log(gasPrice);
};
test().catch();
```

### PR DESCRIPTION
**Pull request by avkos**
## Description

Please include a summary of the changes and be sure to follow our [Contribution Guidelines](../CONTRIBUTIONS.md).

Fixes #6501 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] I have selected the correct base branch.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran `npm run lint` with success and extended the tests and types if necessary.
- [ ] I ran `npm run test:unit` with success.
- [ ] I ran `npm run test:coverage` and my test cases cover all the lines and branches of the added code.
- [ ] I ran `npm run build` and tested `dist/web3.min.js` in a browser.
- [ ] I have tested my code on the live network.
- [ ] I have checked the Deploy Preview and it looks correct.
- [ ] I have updated the `CHANGELOG.md` file in the root folder.
- [ ] I have linked Issue(s) with this PR in "Linked Issues" menu.

### CODE REVIEW

File: packages/web3-types/src/apis/eth_execution_api.ts [hunk 1]
```diff
@@ -104,10 +104,9 @@ export type TransactionUnsignedAPI =
 	| TransactionLegacyUnsignedAPI;
 
 // https://github.com/ethereum/execution-apis/blob/main/src/schemas/transaction.yaml#L211
-export type TransactionSignedAPI =
-	| Transaction1559SignedAPI
-	| Transaction2930SignedAPI
-	| TransactionLegacySignedAPI;
+export type TransactionSignedAPI = Transaction1559SignedAPI &
+	Transaction2930SignedAPI &
+	TransactionLegacySignedAPI;
```
Comment (avkos) [inline packages/web3-types/src/apis/eth_execution_api.ts]: Here is the main fix

File: packages/web3-eth/test/integration/rpc.test.ts [hunk 2]
```diff
@@ -264,6 +264,91 @@ describe('rpc', () => {
 			validateTransaction(res as TransactionInfo);
 			expect(res?.hash).toBe(receipt.transactionHash);
 		});
+		it('check get transaction fields', async () => {
```
Comment (avkos) [inline packages/web3-eth/test/integration/rpc.test.ts:267]: add tests to check fields exists and that no errors to get them for each tx type

File: packages/web3-types/src/apis/eth_execution_api.ts (hunk 1 above)
Comment (Muhammad-Altabba) [inline packages/web3-types/src/apis/eth_execution_api.ts]: I think this type is better like this:
```suggestion
export type TransactionSignedAPI = (
	| Transaction1559SignedAPI
	| Transaction2930SignedAPI
	| TransactionLegacySignedAPI
) |
	Partial<Transaction1559SignedAPI & Transaction2930SignedAPI & TransactionLegacySignedAPI>;
```
 Or what do you all think?

Comment (jdevcs) [inline packages/web3-types/src/apis/eth_execution_api.ts]: According to EL Specs: 
https://ethereum.github.io/execution-apis/api-documentation/ 

`gasPrice` is missing on `Transaction1559SignedAPI`, so only add `gasPrice` in it. 
Result of `getTransactionbyHash` should be `oneOf` 4844 or 2930 or 1559 or legacy tx and not all so it shouldnt be Intersection Type.

image

Comment (jdevcs) [reply packages/web3-types/src/apis/eth_execution_api.ts]: "openrpc": "1.2.4" json generated with [EL Specs](https://github.com/ethereum/execution-apis/tree/057892b256c181232ca22499d1801b9637192698)  have:

```
{
    "name": "eth_getTransactionByHash",
    "summary": "Returns the information about a transaction requested by transaction hash.",
    "params": [
        {
            "name": "Transaction hash",
            "required": true,
            "schema": {
                "title": "32 byte hex value",
                "type": "string",
                "pattern": "^0x[0-9a-f]{64}$"
            }
        }
    ],
    "result": {
        "name": "Transaction information",
        "schema": {
            "oneOf": [
                {
                    "title": "Not Found (null)",
                    "type": "null"
                },
                {
                    "type": "object",
                    "title": "Transaction information",
                    "required": [
                        "blockHash",
                        "blockNumber",
                        "from",
                        "hash",
                        "transactionIndex"
                    ],
                    "unevaluatedProperties": false,
                    "oneOf": [
                        {
                            "title": "Signed 4844 Transaction",
                            "type": "object",
                            "required": [
                                "accessList",
                                "blobVersionedHashes",
                                "chainId",
                                "gas",
                                "input",
                                "maxFeePerBlobGas",
                                "maxFeePerGas",
                                "maxPriorityFeePerGas",
                                "nonce",
                                "r",
                                "s",
                                "to",
                                "type",
                                "value",
                                "yParity"
                            ],
                            "properties": {
                                "type": {
                                    "title": "type",
                                    "type": "string",
                                    "pattern": "^0x([0-9,a-f,A-F]?){1,2}$"
                                },
                                "nonce": {
                                    "title": "nonce",
                                    "type": "string",
                                    "pattern": "^0x([1-9a-f]+[0-9a-f]*|0)$"
                                },
                                "to": {
                                    "title": "to address",
                                    "type": "string",
                                    "pattern": "^0x[0-9,a-f,A-F]{40}$"
                                },
                                "gas": {
                                    "title": "gas limit",
                                    "type": "string",
                                    "pattern": "^0x([1-9a-f]+[0-9a-f]*|0)$"
                                },
                                "value": {
                                    "title": "value",
                                    "type": "string",
                                    "pattern": "^0x([1-9a-f]+[0-9a-f]*|0)$"
                                },
                                "input": {
                                    "title": "input data",
                                    "type": "string",
                                    "pattern": "^0x[0-9a-f]*$"
                                },
                                "maxPriorityFeePerGas": {
                                    "title": "max priority fee per gas",
                                    "type": "string",
                                    "pattern": "^0x([1-9a-f]+[0-9a-f]*|0)$",
                                    "description": "Maximum fee per gas the sender is willing to pay to miners in wei"
                                },
                                "maxFeePerGas": {
                                    "title": "max fee per gas",
                                    "type": "string",
                                    "pattern": "^0x([1-9a-f]+[0-9a-f]*|0)$",
                                    "description": "The maximum total fee per gas the sender is willing to pay (includes the network / base fee and miner / priority fee) in wei"
                                },
                                "maxFeePerBlobGas": {
                                    "title": "max fee per blob gas",
                                    "type": "string",
                                    "pattern": "^0x([1-9a-f]+[0-9a-f]*|0)$",
                                    "description": "The maximum total fee per gas the sender is willing to pay for blob gas in wei"
                                },
                                "accessList": {
                                    "title": "accessList",
                                    "type": "array",
                                    "description": "EIP-2930 access list",
                                    "items": {
                                        "title": "Access list entry",
                                        "type": "object",
                                        "additionalProperties": false,
                                        "properties": {
                                            "address": {
                                                "title": "hex encoded address",
                                                "type": "string",
                                                "pattern": "^0x[0-9,a-f,A-F]{40}$"
                                            },
                                            "storageKeys": {
                                                "type": "array",
                                                "items": {
                                                    "title": "32 byte hex value",
                                                    "type": "string",
                                                    "pattern": "^0x[0-9a-f]{64}$"
                                                }
                                            }
                                        }
                                    }
                                },
                                "blobVersionedHashes": {
                                    "title": "blobVersionedHashes",
                                    "description": "List of versioned blob hashes associated with the transaction's EIP-4844 data blobs.",
                                    "type": "array",
                                    "items": {
                                        "title": "32 byte hex value",
                                        "type": "string",
                                        "pattern": "^0x[0-9a-f]{64}$"
                                    }
                                },
                                "chainId": {
                                    "title": "chainId",
                                    "type": "string",
                                    "pattern": "^0x([1-9a-f]+[0-9a-f]*|0)$",
                                    "description": "Chain ID that this transaction is valid on."
                                },
                                "yParity": {
                                    "title": "yParity",
                                    "type": "string",
                                    "pattern": "^0x([1-9a-f]+[0-9a-f]*|0)$",
                                    "description": "The parity (0 for even, 1 for odd) of the y-value of the secp256k1 signature."
                                },
                                "r": {
                                    "title": "r",
                                    "type": "string",
                                    "pattern": "^0x([1-9a-f]+[0-9a-f]*|0)$"
                                },
                                "s": {
                                    "title": "s",
                                    "type": "string",
                                    "pattern": "^0x([1-9a-f]+[0-9a-f]*|0)$"
                                }
                            }
                        },
                        {
                            "title": "Signed 1559 Transaction",
                            "type": "object",
                            "required": [
                                "accessList",
                                "chainId",
                                "gas",
                                "gasPrice",
                                "input",
                                "maxFeePerGas",
                                "maxPriorityFeePerGas",
                                "nonce",
                                "r",
                                "s",
                                "type",
                                "value",
                                "yParity"
                            ],
                            "properties": {
                                "type": {
                                    "title": "type",
                                    "type": "string",
                                    "pattern": "^0x2$"
                                },
                                "nonce": {
                                    "title": "nonce",
                                    "type": "string",
                                    "pattern": "^0x([1-9a-f]+[0-9a-f]*|0)$"
                                },
                                "to": {
                                    "title": "to address",
                                    "oneOf": [
                                        {
                                            "title": "Contract Creation (null)",
                                            "type": "null"
                                        },
                                        {
                                            "title": "Address",
                                            "type": "string",
                                            "pattern": "^0x[0-9,a-f,A-F]{40}$"
                                        }
                                    ]
                                },
                                "gas": {
                                    "title": "gas limit",
                                    "type": "string",
                                    "pattern": "^0x([1-9a-f]+[0-9a-f]*|0)$"
                                },
                                "value": {
                                    "title": "value",
                                    "type": "string",
                                    "pattern": "^0x([1-9a-f]+[0-9a-f]*|0)$"
                                },
                                "input": {
                                    "title": "input data",
                                    "type": "string",
                                    "pattern": "^0x[0-9a-f]*$"
                                },
                                "maxPriorityFeePerGas": {
                                    "title": "max priority fee per gas",
                                    "type": "string",
                                    "pattern": "^0x([1-9a-f]+[0-9a-f]*|0)$",
                                    "description": "Maximum fee per gas the sender is willing to pay to miners in wei"
                                },
                                "maxFeePerGas": {
                                    "title": "max fee per gas",
                                    "type": "string",
                                    "pattern": "^0x([1-9a-f]+[0-9a-f]*|0)$",
                                    "description": "The maximum total fee per gas the sender is willing to pay (includes the network / base fee and miner / priority fee) in wei"
                                },
                                "gasPrice": {
                                    "title": "gas price",
                                    "type": "string",
                                    "pattern": "^0x([1-9a-f]+[0-9a-f]*|0)$",
                                    "description": "The effective gas price paid by the sender in wei. For transactions not yet included in a block, this value should be set equal to the max fee per gas. This field is DEPRECATED, please transition to using effectiveGasPrice in the receipt object going forward."
                                },
                                "accessList": {
                                    "title": "accessList",
                                    "type": "array",
                                    "description": "EIP-2930 access list",
                                    "items": {
                                        "title": "Access list entry",
                                        "type": "object",
                                        "additionalProperties": false,
                                        "properties": {
                                            "address": {
                                                "title": "hex encoded address",
                                                "type": "string",
                                                "pattern": "^0x[0-9,a-f,A-F]{40}$"
                                            },
                                            "storageKeys": {
                                                "type": "array",
                                                "items": {
                                                    "title": "32 byte hex value",
                                                    "type": "string",
                                                    "pattern": "^0x[0-9a-f]{64}$"
                                                }
                                            }
                                        }
                                    }
                                },
                                "chainId": {
                                    "title": "chainId",
                                    "type": "string",
                                    "pattern": "^0x([1-9a-f]+[0-9a-f]*|0)$",
                                    "description": "Chain ID that this transaction is valid on."
                                },
                                "yParity": {
                                    "title": "yParity",
                                    "type": "string",
                                    "pattern": "^0x([1-9a-f]+[0-9a-f]*|0)$",
                                    "description": "The parity (0 for even, 1 for odd) of the y-value of the secp256k1 signature."
                                },
                                "v": {
                                    "title": "v",
                                    "type": "string",
                                    "pattern": "^0x([1-9a-f]+[0-9a-f]*|0)$",
                                    "description": "For backwards compatibility, `v` is optionally provided as an alternative to `yParity`. This field is DEPRECATED and all use of it should migrate to `yParity`."
                                },
                                "r": {
                                    "title": "r",
                                    "type": "string",
                                    "pattern": "^0x([1-9a-f]+[0-9a-f]*|0)$"
                                },
                                "s": {
                                    "title": "s",
                                    "type": "string",
                                    "pattern": "^0x([1-9a-f]+[0-9a-f]*|0)$"
                                }
                            }
                        },
                        {
                            "title": "Signed 2930 Transaction",
                            "type": "object",
                            "required": [
                                "accessList",
                                "chainId",
                                "gas",
                                "gasPrice",
                                "input",
                                "nonce",
                                "r",
                                "s",
                                "type",
                                "value",
                                "yParity"
                            ],
                            "properties": {
                                "type": {
                                    "title": "type",
                                    "type": "string",
                                    "pattern": "^0x1$"
                                },
                                "nonce": {
                                    "title": "nonce",
                                    "type": "string",
                                    "pattern": "^0x([1-9a-f]+[0-9a-f]*|0)$"
                                },
                                "to": {
                                    "title": "to address",
                                    "oneOf": [
                                        {
                                            "title": "Contract Creation (null)",
                                            "type": "null"
                                        },
                                        {
                                            "title": "Address",
                                            "type": "string",
                                            "pattern": "^0x[0-9,a-f,A-F]{40}$"
                                        }
                                    ]
                                },
                                "gas": {
                                    "title": "gas limit",
                                    "type": "string",
                                    "pattern": "^0x([1-9a-f]+[0-9a-f]*|0)$"
                                },
                                "value": {
                                    "title": "value",
                                    "type": "string",
                                    "pattern": "^0x([1-9a-f]+[0-9a-f]*|0)$"
                                },
                                "input": {
                                    "title": "input data",
                                    "type": "string",
                                    "pattern": "^0x[0-9a-f]*$"
                                },
                                "gasPrice": {
                                    "title": "gas price",
                                    "type": "string",
                                    "pattern": "^0x([1-9a-f]+[0-9a-f]*|0)$",
                                    "description": "The gas price willing to be paid by the sender in wei"
                                },
                                "accessList": {
                                    "title": "accessList",
                                    "type": "array",
                                    "description": "EIP-2930 access list",
                                    "items": {
                                        "title": "Access list entry",
                                        "type": "object",
                                        "additionalProperties": false,
                                        "properties": {
                                            "address": {
                                                "title": "hex encoded address",
                                                "type": "string",
                                                "pattern": "^0x[0-9,a-f,A-F]{40}$"
                                            },
                                            "storageKeys": {
                                                "type": "array",
                                                "items": {
                                                    "title": "32 byte hex value",
                                                    "type": "string",
                                                    "pattern": "^0x[0-9a-f]{64}$"
                                                }
                                            }
                                        }
                                    }
                                },
                                "chainId": {
                                    "title": "chainId",
                                    "type": "string",
                                    "pattern": "^0x([1-9a-f]+[0-9a-f]*|0)$",
                                    "description": "Chain ID that this transaction is valid on."
                                },
                                "yParity": {
                                    "title": "yParity",
                                    "type": "string",
                                    "pattern": "^0x([1-9a-f]+[0-9a-f]*|0)$",
                                    "description": "The parity (0 for even, 1 for odd) of the y-value of the secp256k1 signature."
                                },
                                "v": {
                                    "title": "v",
                                    "type": "string",
                                    "pattern": "^0x([1-9a-f]+[0-9a-f]*|0)$",
                                    "description": "For backwards compatibility, `v` is optionally provided as an alternative to `yParity`. This field is DEPRECATED and all use of it should migrate to `yParity`."
                                },
                                "r": {
                                    "title": "r",
                                    "type": "string",
                                    "pattern": "^0x([1-9a-f]+[0-9a-f]*|0)$"
                                },
                                "s": {
                                    "title": "s",
                                    "type": "string",
                                    "pattern": "^0x([1-9a-f]+[0-9a-f]*|0)$"
                                }
                            }
                        },
                        {
                            "title": "Signed Legacy Transaction",
                            "type": "object",
                            "required": [
                                "gas",
                                "gasPrice",
                                "input",
                                "nonce",
                                "r",
                                "s",
                                "type",
                                "v",
                                "value"
                            ],
                            "properties": {
                                "type": {
                                    "title": "type",
                                    "type": "string",
                                    "pattern": "^0x0$"
                                },
                                "nonce": {
                                    "title": "nonce",
                                    "type": "string",
                                    "pattern": "^0x([1-9a-f]+[0-9a-f]*|0)$"
                                },
                                "to": {
                                    "title": "to address",
                                    "oneOf": [
                                        {
                                            "title": "Contract Creation (null)",
                                            "type": "null"
                                        },
                                        {
                                            "title": "Address",
                                            "type": "string",
                                            "pattern": "^0x[0-9,a-f,A-F]{40}$"
                                        }
                                    ]
                                },
                                "gas": {
                                    "title": "gas limit",
                                    "type": "string",
                                    "pattern": "^0x([1-9a-f]+[0-9a-f]*|0)$"
                                },
                                "value": {
                                    "title": "value",
                                    "type": "string",
                                    "pattern": "^0x([1-9a-f]+[0-9a-f]*|0)$"
                                },
                                "input": {
                                    "title": "input data",
                                    "type": "string",
                                    "pattern": "^0x[0-9a-f]*$"
                                },
                                "gasPrice": {
                                    "title": "gas price",
                                    "type": "string",
                                    "pattern": "^0x([1-9a-f]+[0-9a-f]*|0)$",
                                    "description": "The gas price willing to be paid by the sender in wei"
                                },
                                "chainId": {
                                    "title": "chainId",
                                    "type": "string",
                                    "pattern": "^0x([1-9a-f]+[0-9a-f]*|0)$",
                                    "description": "Chain ID that this transaction is valid on."
                                },
                                "v": {
                                    "title": "v",
                                    "type": "string",
                                    "pattern": "^0x([1-9a-f]+[0-9a-f]*|0)$"
                                },
                                "r": {
                                    "title": "r",
                                    "type": "string",
                                    "pattern": "^0x([1-9a-f]+[0-9a-f]*|0)$"
                                },
                                "s": {
                                    "title": "s",
                                    "type": "string",
                                    "pattern": "^0x([1-9a-f]+[0-9a-f]*|0)$"
                                }
                            }
                        }
                    ],
                    "properties": {
                        "blockHash": {
                            "title": "block hash",
                            "type": "string",
                            "pattern": "^0x[0-9a-f]{64}$"
                        },
                        "blockNumber": {
                            "title": "block number",
                            "type": "string",
                            "pattern": "^0x([1-9a-f]+[0-9a-f]*|0)$"
                        },
                        "from": {
                            "title": "from address",
                            "type": "string",
                            "pattern": "^0x[0-9,a-f,A-F]{40}$"
                        },
                        "hash": {
                            "title": "transaction hash",
                            "type": "string",
                            "pattern": "^0x[0-9a-f]{64}$"
                        },
                        "transactionIndex": {
                            "title": "transaction index",
                            "type": "string",
                            "pattern": "^0x([1-9a-f]+[0-9a-f]*|0)$"
                        }
                    }
                }
            ]
        }
    }
},
```

Comment (jdevcs) [reply packages/web3-types/src/apis/eth_execution_api.ts]: changes required in this as suggested.

Comment (avkos) [reply packages/web3-types/src/apis/eth_execution_api.ts]: yes. it was `or` between types, and looks like it is how it should be
```export type TransactionSignedAPI = 
        | Transaction1559SignedAPI
	| Transaction2930SignedAPI 
	| TransactionLegacySignedAPI
```

But in this case `TransactionSignedAPI` is always `Transaction1559SignedAPI`, and It doesn't matter which type or parameters an object has.

Comment (avkos) [reply packages/web3-types/src/apis/eth_execution_api.ts]: it was changed yesterday

**Review by jdevcs — APPROVED** [review]
also update changelog

File: packages/web3-eth/src/rpc_method_wrappers.ts [hunk 3]
```diff
@@ -740,7 +740,7 @@ export async function signTransaction<ReturnFormat extends DataFormat>(
 ) {
 	const response = await ethRpcMethods.signTransaction(
 		web3Context.requestManager,
-		formatTransaction(transaction, ETH_DATA_FORMAT),
+		formatTransaction(transaction as TransactionWithSenderAPI, ETH_DATA_FORMAT),
```
Comment (jdevcs) [inline packages/web3-eth/src/rpc_method_wrappers.ts]: we dnt need to cast it here, as issue under discussion was for `getTransaction` and other functions in this file @avkos

Comment (avkos) [reply packages/web3-eth/src/rpc_method_wrappers.ts]: reverted

**Review by luu-alex — APPROVED** [review]

**Pull request merged**